### PR TITLE
Preyansh/arp

### PR DESCRIPTION
**Pull request by kingpreyansh**
- worked on creating the functionality to handle the ARP req
- populating and sending the ARP reply packet

### CODE REVIEW

File: cs144_lab3/router/sr_router.c [hunk 1]
```diff
@@ -31,7 +31,6 @@
  *---------------------------------------------------------------------*/
 
 void sr_handle_IP_packet(struct sr_instance * sr, uint8_t * packet, unsigned int len, char *);
```
Comment (alinabuzila) [inline cs144_lab3/router/sr_router.c:33]: we need to remove sr and .o files it will give merge conflicts

Comment (kingpreyansh) [reply cs144_lab3/router/sr_router.c:33]: i'll remove these

File: cs144_lab3/router/sr_router.c [hunk 2]
```diff
@@ -111,24 +108,8 @@ void sr_handlepacket(struct sr_instance* sr,
 
 }
 
-
+/* TODO: Ask Alina about how she's going to deal with the IP forwarding */
```
Comment (alinabuzila) [inline cs144_lab3/router/sr_router.c:111]: rebase pls

File: cs144_lab3/router/sr_router.c [hunk 3]
```diff
@@ -143,50 +124,99 @@ void sr_handle_IP_packet(struct sr_instance *sr, uint8_t *packet, unsigned int l
...
-    fprintf(stderr, "[Error] Unknown ARP op-code/operation: %d", operation);
-  }
+    for (i = 0; i < ETHER_ADDR_LEN; i++) {
+        arp_header->ar_tha[i] = req_arp_hdr->ar_sha[i];
+        arp_header->ar_sha[i] = interface->addr[i];
+    }
+    
+    arp_header->ar_sip = interface->ip;
+    arp_header->ar_tip = req_arp_hdr->ar_sip;
+    sr_send_packet(sr_instance, reply_packet, reply_length, interface->name);
```
Comment (alinabuzila) [inline cs144_lab3/router/sr_router.c:200]: must free the pointer you allocated space

**Pull request merged**